File: LeetCode/JavaScript/#506 Relative Ranks.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var findRelativeRanks = function(nums) {
    var temp = [];
    var len = nums.length;
    var i,pos;

    for(var k = 0;k<len;k++){
      temp[k] = nums[k];
    }
    temp = temp.sort(function(a,b){return b-a;});
    for(i=0;i<len;i++){
      pos = nums.indexOf(temp[i]);
      if(i===0){
        nums[pos] = "Gold Medal";
      }else if(i===1){
        nums[pos] = "Silver Medal";
      }else if(i===2){
        nums[pos] = "Bronze Medal";
      }else{
        nums[pos] = ""+(i+1);
      }
    }
    return nums;
};

File: LeetCode/Python/#506 Relative Ranks.py
class Solution(object):
    def findRelativeRanks(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        s = {n: i for i,n in enumerate(sorted(nums,reverse=True))}

        medals = ["Gold","Silver","Bronze"]
        for n in range(len(nums)):
            if s[nums[n]]>=len(medals):
                nums[n] = str(s[nums[n]]+1)
            else:
                nums[n] = (medals[s[nums[n]]]+" Medal")

        return nums


# def findRelativeRanks(nums):
#     """
#     :type nums: List[int]
#     :rtype: List[str]
#     """
#     s = {n: i for i,n in enumerate(sorted(nums,reverse=True))}
#
#     medals = ["Gold","Silver","Bronze"]
#     for n in range(len(nums)):
#         if s[nums[n]]>=len(medals):
#             nums[n] = str(s[nums[n]]+1)
#         else:
#             nums[n] = (medals[s[nums[n]]]+" Medal")
#
#     print(nums)
#     return nums
#
# findRelativeRanks([1,2,3,4,5])
# findRelativeRanks([3,1,4,5,2])